src/javascripts/templates/template.js

```diff
@@ -0,0 +1,70 @@
+import { formatNumber } from "../helpers/helpers"
+
+class Template {
+    /**
+     * Constructor of Template object
+     */
+    constructor(){ }
+
+     /**
+     * HTML Template for render an Transaction object
+     * @param {object} transaction 
+     */
+
+    static transaction = (transaction) => `
```

Comment:
@HoangLiarPhuongLe You should reorganize this pattern into a function to improve readability and make it easier to include conditional logic and calculations. For example:
```
static transaction = (transaction) => {
    const transactionDate = new Date(transaction.date);
    const isIncome = transaction.outflow >= 0;
    const amountSign = isIncome ? '+' : '-';
    const amountClass = isIncome ? 'income' : 'outflow';

    return `
      <div class="transaction-detail">
        <div class="transaction-info">
          <div class="datetime">
            <p class="date-text">${transactionDate.getDate()}</p>
            <div>
              <div class="daymonthyear">
                <p class="daymonthyear-text">${transactionDate.toLocaleDateString('en-US', { weekday: 'long' })},</p>
                <p class="daymonthyear-text">${transactionDate.toLocaleDateString('en-US', { month: 'long' })}</p>
                <p class="daymonthyear-text">${transactionDate.getFullYear()}</p>
              </div>
              <p class="note-text">${transaction.note}</p>
            </div>
          </div>
          <p class="amount-${amountClass}">${amountSign}$ ${formatNumber(Math.abs(transaction.outflow))}</p>
        </div>
      </div>
    `;
  }
```

The same for `category`